tracer/test/Datadog.Trace.Tests/DataStreamsMonitoring/DataStreamsWriterTests.cs

```diff
@@ -76,14 +112,32 @@ public async Task WritesAStatsPoint_OnClose()
         api.Sent.Should().ContainSingle();
     }
 
+    [Theory]
+    [InlineData(false)]
+    [InlineData(null)]
+    public async Task WhenNotSupported_DoesNotWriteAStatsPoint_OnClose(bool? isSupported)
+    {
+        var bucketDuration = 100_000_000; // 100 ms
+        var api = new StubApi();
+        var writer = CreateWriter(api, out var discovery, bucketDuration);
+        TriggerSupportUpdate(discovery, isSupported);
+
+        writer.Add(CreateStatsPoint());
+
+        await writer.DisposeAsync();
+
+        api.Sent.Should().BeEmpty();
+    }
+
     [Fact]
-    public async Task AllBucketsAreReportedOnClose()
+    public async Task WhenSupported_AllBucketsAreReportedOnClose()
     {
         // using a very long duration here so that we're guaranteed to still
         // be in the "current" bucket when we call flush and close
         var bucketDuration = int.MaxValue; // 100 s
```

Comment:
nitpicky (and technically outside of the PR)

```suggestion
        var bucketDuration = int.MaxValue;
```